src/server/engine_shard_set.cc

```diff
@@ -75,6 +160,8 @@ EngineShard::EngineShard(util::ProactorBase* pb, bool update_db_time, mi_heap_t*
   tmp_str1 = sdsempty();
 
   db_slice_.UpdateExpireBase(absl::GetCurrentTimeNanos() / 1000000, 0);
+
+  StartDefragTask(pb);
```

Comment:
maybe not start but Schedule or Register ?

Reply:
Schedule it is